src/Order.sol

```diff
@@ -161,6 +165,66 @@ contract Order {
         emit ActiveToggled(active);
     }
 
+    /// @dev returns buyer refund per unit for offer
+    function refundPerUnit(Offer memory offer)
+        internal
+        view
+        virtual
+        returns (uint256)
+    {
+        if (offer.pricePerUnit > offer.costPerUnit) {
+            return offer.pricePerUnit - offer.costPerUnit;
+        } else {
+            return 0;
+        }
+    }
+
+    /// @dev returns buyer stake per unit for offer 
+    function buyerStakePerUnit(Offer memory offer)
+        internal
+        view
+        virtual
+        returns (uint256)
+    {
+        if (offer.costPerUnit > offer.pricePerUnit) {
+            return offer.costPerUnit - offer.pricePerUnit;
+        } else {
+            return 0;
+        }
+    }
+
+    /// @dev returns buyer for offer with given taker
+    function buyer(address taker_) 
```

Comment:
I think we had a convention of `variable_` when `variable` is already a member of the struct, but `taker` is not a member of `Order`. I see you've added `taker_` in a few other locations as well.

To be fair, I think the other way around is less error-prone (i.e. trailing underscore indicates explicit access to a member variable, otherwise it's a local variable). But we should at least keep it consistent.